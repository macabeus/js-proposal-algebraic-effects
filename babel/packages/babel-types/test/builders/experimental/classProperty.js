import * as t from "../../..";

describe("builders", function() {
  describe("experimental", function() {
    describe("classProperty", function() {
      it("should validate", function() {
        expect(
          t.classProperty(t.stringLiteral("test"), t.numericLiteral(1)),
        ).toMatchSnapshot();

        expect(
          t.classProperty(
            t.stringLiteral("test"),
            t.numericLiteral(1),
            null,
            null,
            false,
            true,
          ),
        ).toMatchSnapshot();
      });
    });
  });
});
