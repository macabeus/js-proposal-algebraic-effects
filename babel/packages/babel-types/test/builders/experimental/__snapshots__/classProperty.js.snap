// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders experimental classProperty should validate 1`] = `
Object {
  "computed": false,
  "decorators": null,
  "key": Object {
    "type": "StringLiteral",
    "value": "test",
  },
  "static": false,
  "type": "ClassProperty",
  "typeAnnotation": null,
  "value": Object {
    "type": "NumericLiteral",
    "value": 1,
  },
}
`;

exports[`builders experimental classProperty should validate 2`] = `
Object {
  "computed": false,
  "decorators": null,
  "key": Object {
    "type": "StringLiteral",
    "value": "test",
  },
  "static": true,
  "type": "ClassProperty",
  "typeAnnotation": null,
  "value": Object {
    "type": "NumericLiteral",
    "value": 1,
  },
}
`;
